src/Common/Menus/Abstract_Submenu.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * An EXAMPLE class modeling a submenu - you should not extend this.
+ * Instead, you extend the base Abstract_Menu class and add the Submenu Trait!
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus;
+
+use TEC\Common\Menus\Traits\Submenu;
+
+/**
+ * Class Menu
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Menus
+ */
+abstract class Submenu_Example extends Abstract_Menu implements Menu_Contract {
```

Comment:
This is where I see the noise coming up: we have Menu pages and sub-menu pages. They are different things requiring different logic. I think the extension/inheritance/implementation stack here could be cut down having a base `abstract` class for menu pages and a base `abstract` class for sub-menu pages.

Reply:
Did you read the docblock?

> Are we re-doing what we recently introduced here? https://github.com/the-events-calendar/tribe-common/blob/master/src/Tribe/Admin/Pages.php
> 
> We introduced that as part of the Tickets menu, and all plugins are using now that "framework" to already register and manage admin pages.

@juanfra - yes. the idea behind this is something that is less opinionated, less tied to TEC/ET, does not include admin pages in it automatically, and can be separated out and shared both with Stellar and with the rest of the world.

Reply:
Not sure why GitHub decided to attach that to this comment 🤦🏼

Reply:
@Camwyn I don't understand the opinionated part. Could you please expand?

I feel that menus and submenus are just pages, thinking abstractly. Creating whole new classes for something that was already done a couple months ago (and that's ~300 lines) would be maybe taking valuable time from us. To add up, we are still using `tec_` prefixes, so I don't understand how this would be less tied to TEC/ET.

With the current admin pages implementation there's no page generated, that was moved to the plugins, and it was kind of one of the goals, because for ET as standalone the TEC page was created automatically from common.

Reply:
The final home for this would be a public Stellar library - so not `tec_` prefixed (just convenient for now) and not built with TEC/ET's needs specifically in mind.

Note it uses no `tribe()` functionality (although that would be convenient - it won't, come time to separate it)